src/python/pants/rules/core/binary.py

```diff
@@ -113,18 +113,18 @@ class Binary(Goal):
     registered_target_types: RegisteredTargetTypes,
 ) -> CreatedBinary:
     target = wrapped_target.target
-    binary_config_types: Iterable[Type[BinaryConfiguration]] = union_membership.union_rules[
+    config_types: Iterable[Type[BinaryConfiguration]] = union_membership.union_rules[
```

Comment:
Name change to make this Configuration code a little more generic. I'm starting to look for patterns that make sense to be factored up.